File: Level1/ValidPalindrome.java
package Level1;

public class ValidPalindrome {    

    public static void main(String[] args) {
       System.out.println(isPalindrome("A man, a plan, a canal: Panama")); 
    }

    
    public static boolean isPalindrome(String s) {
        boolean ans = true ;
        // replace all non-Charaters
        s = s.toLowerCase().replaceAll("[^A-Za-z0-9]", "");
        // compare i, j index
        for(int i=0, j=s.length()-1 ;i <= j; i++, j--){
           if(s.charAt(i) != s.charAt(j) ){
                return false;
           }
        }

        return ans;
    }
    
}

File: two_pointers/valid_palindrome.py
import re


class Solution:
    def isPalindrome(self, s: str) -> bool:
        # data = re.sub(r'[^0-9a-z]','', s.lower())
        # return data == data[::-1]
        pass


sol = Solution().isPalindrome("A man, a plan, a canal: Panama")
print(sol)